### PR TITLE
[BUG FIX] Fixing Async Consumer Logic to work based on SelectConnection

### PR DESCRIPTION
**Pull request by NeonKirill**
# Description
Async Consumer based on `aio-pika` dependency required a lot of extra effort from code refactoring and async behaviour maintance.

The proposal here is to get rid of it, and using SelectConnection which has same API interface and seamless callback-based async behavior.

### CODE REVIEW

File: neon_mq_connector/consumers/select_consumer.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
+import threading
```
Comment (NeonDaniel) [inline neon_mq_connector/consumers/select_consumer.py:1]: New file needs license notice

File: neon_mq_connector/consumers/__init__.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+__all__ = [
```
Comment (NeonDaniel) [inline neon_mq_connector/consumers/__init__.py:1]: New file needs license notice

**Review by NeonDaniel — CHANGES_REQUESTED** [review]
Few small license and backwards-compat. notes. I also still need to review the parameterized test behavior as I'm not very familiar with that functionality in Pytest (yet)

File: neon_mq_connector/consumers/blocking_consumer.py [hunk 3]
```diff
@@ -8,20 +8,21 @@
 from neon_mq_connector.utils import consumer_utils
 
 
-class ConsumerThread(threading.Thread):
```
[the remark this answers is not in the document]
Comment (NeonDaniel) [reply neon_mq_connector/consumers/blocking_consumer.py:11]: I see the reference in `neon_mq_connector.consumer` reference was new in #99 and the `neon_mq_connector.connector` backwards-compat import remains, so this should be a non-issue

File: tests/test_backward_compatibility.py [hunk 4]
```diff
@@ -62,21 +63,22 @@ def setUpClass(cls) -> None:
...
+                BlockingConsumerThread,
             )
         ]
     )
     def test_stable_register_consumer_args(
         self,
-        test_name: str,
-        async_mode_enabled: bool,
-        expected_consumer_instance
+        test_name,
```
Comment (NeonDaniel) [inline tests/test_backward_compatibility.py:77]: From [parameterized docs on PyPI](https://pypi.org/project/parameterized/), should this be `name`?

Is there a simple method for confirming that both test cases are being run in GHA? I only see that each test case passed, not anything to indicate the different parameterized inputs were tested.

Comment (NeonKirill) [reply tests/test_backward_compatibility.py:77]: > From [parameterized docs on PyPI](https://pypi.org/project/parameterized/), should this be `name`?
> 
> Is there a simple method for confirming that both test cases are being run in GHA? I only see that each test case passed, not anything to indicate the different parameterized inputs were tested.

Yeah, we may use either `name` or `test_name`, both are resolved correctly by the parameterized. You may download and lookup backward compatibility test artifacts for any python version there is a note that both of those tests ran and succeeded.

This is sample artifact for Python 3.10:

`<?xml version="1.0" encoding="utf-8"?><testsuites><testsuite name="pytest" errors="0" failures="0" skipped="0" tests="2" time="0.760" timestamp="2024-11-28T19:51:41.294575" hostname="fv-az1427-607">
        <testcase classname="tests.test_backward_compatibility.TestBackwardCompatibility" name="test_stable_register_consumer_args_0_async_mode_enabled" time="0.060" />
         <testcase classname="tests.test_backward_compatibility.TestBackwardCompatibility" name="test_stable_register_consumer_args_1_async_mode_disabled" time="0.442" /></testsuite></testsuites>`

**Review by NeonDaniel — APPROVED** [review]

**Pull request merged**